### DISCUSSION TITLE
List of Gcode Commands

### DISCUSSION THREAD

**Discussion by Abowyer-Mike** [terjeio/ioSender#401] [Q&A] [answered]
Total Noob so forgive my ignorance.  After hours of scouring the internet and the discussions here, I'm reluctantly asking this question.  I'm trying to make macros in iosender to switch the Aux00, Aux01, Aux02, and Flood on an E5XMCS controller and can't find documentation any where.  Any help would be appreciated.  BTW, thank you very much TERJEIO for putting this together.  One can't imagine the hours you have sunk into this project.

Mike

**Comment by phil-barrett** [comment, marked as answer]
Assuming that your controller has pins for those, there are several M codes you would use.  

[M7-9 controls flood and mist](https://linuxcnc.org/docs/html/gcode/m-code.html#mcode:m7-m8-m9) - To turn on flood, use M8 (off, M9).  M7 for mist. 

[M62-M65 controls Aux output](https://linuxcnc.org/docs/html/gcode/m-code.html#mcode:m62-m65).  For example M64 P0 will turn on Aux00 immediately.  M65 P0 will turn it off.

**Reply by Abowyer-Mike** [reply]
Thank you, Mr. Barrett.  That was indeed correct.  I subsequently found all of the documentation I needed here:  https://makerhardware.net/knowledge-base/electronics/e5x-mcs-t4-1-controller/  
And the commands that you replied with map well with it.  
Cheers, 
Mike
